src/DateTime.cc

```diff
@@ -339,6 +339,8 @@ DateTime::DateTime(int year, int month, int day, int hr, int min, int sec, Times
     tm.tm_mday = day;
     tm.tm_hour = hr;
     tm.tm_min = min;
+    bool fix60 = sec == 60;
+    if (fix60) sec = 59;
```

Comment:
I suggest not allowing sec == 0 unless this is UTC. Also please consider using the same variable name (inleap) as elsewhere in the code.

If the time system is TAI, for example, this code causes different behavior for different invalid inputs (sec=60 vs sec=61), which could make debugging errors more difficult.